wire-ios-data-model/Tests/Source/Model/Observer/ConversationObserverTests.swift

```diff
@@ -940,7 +941,9 @@ final class ConversationObserverTests: NotificationDispatcherTestBase {
 
         let modifier: (ZMConversation, ConversationObserver) -> Void = { _, _ in
             self.performPretendingUiMocIsSyncMoc {
-                legalHoldClient.deleteClientAndEndSession()
+                // Can't call async function inside the synchronous modifier block.
+                // We need an async version of `checkThatItNotifiesTheObserverOfAChange`
+                // legalHoldClient.deleteClientAndEndSession()
```

Comment:
Can't we use a `WaitingGroupTask` and `testWaitForAllGroupsToBeEmpty`?

Reply:
Maybe I'm not saying it's impossible we still need to re-write `checkThatItNotifiesTheObserverOfAChange` or rather an additional async version.